### DISCUSSION TITLE
Will using rust as a scripting language increase the threshold？

### DISCUSSION THREAD

**Discussion by C-BJ** [bevyengine/bevy#4338] [Q&A] [answered]
I am new here，Will using rust as a scripting language increase the threshold？

**Comment by bjorn3** [comment]
What threshold?

**Reply by C-BJ** [reply]
> What threshold?

For beginners and people who do game art or planning

Compared with Lua and Wren or other similar scripting languages

**Comment by james7132** [comment, marked as answer]
Compared to your typical introductory game engines like Unity, Unreal, or Godot? Probably-ish. For absolute beginners, Rust has traditionally put a lot of the learning curve up front, and that's not too viable as a baseline introductory language to programming. It's by far one of the most prevalent gripes people have with learning Rust as a whole, even for experienced programmers.

However, when you compare Bevy's ease of use to the other options in the same niche, like Unity's DOTS/ECS, Bevy's rather easy domain specific language for handling systems and components makes it rather easy to onboard onto, provided a minimal amount of prior programming experience. In fact, it's become quite common to hear of users trying out Bevy as a way to learn Rust, which I personally think is a good sign that the overall API design is doing it's job in easing onboarding into the engine.

If you want a comparison as to how boilerplate and concept heavy doing ECS development can be, just take a look at the simplest example taken straight from Unity's ECS examples repo:

```csharp
using Unity.Entities;
using Unity.Jobs;
using Unity.Mathematics;
using Unity.Transforms;

// This system updates all entities in the scene with both a RotationSpeed_ForEach and Rotation component.

// ReSharper disable once InconsistentNaming
public partial class RotationSpeedSystem_ForEach : SystemBase
{
    // OnUpdate runs on the main thread.
    protected override void OnUpdate()
    {
        float deltaTime = Time.DeltaTime;

        // Schedule job to rotate around up vector
        Entities
            .WithName("RotationSpeedSystem_ForEach")
            .ForEach((ref Rotation rotation, in RotationSpeed_ForEach rotationSpeed) =>
            {
                rotation.Value = math.mul(
                    math.normalize(rotation.Value),
                    quaternion.AxisAngle(math.up(), rotationSpeed.RadiansPerSecond * deltaTime));
            })
            .ScheduleParallel();
    }
}
```

**Reply by C-BJ** [reply]
Can the bevy editor in the roadmap let people who do game art (planning) not use rust

**Reply by C-BJ** [reply]
For example, make some settings for shaders

**Reply by NiklasEi** [reply]
I think there can be quite some work done on a Bevy game without touching Rust code. Game art and shaders are assets and should not live in your code. Editing these files does not require any Rust knowledge.

**Reply by colepoirier** [reply]
> Can the bevy editor in the roadmap let people who do game art (planning) not use rust

I think that is the plan :)

Bevy as an engine will always be code first, but the Bevy *editor* has the goal of being useful to production game studios which IIUC absolutely need the ability for their non-programmer team members to work without touching or knowing code. Rest assured the Bevy devs are keeping your use case in mind as they discuss the plans for the editor!

**Reply by C-BJ** [reply]
> > Can the bevy editor in the roadmap let people who do game art (planning) not use rust
> 
> I think that is the plan :)
> 
> Bevy as an engine will always be code first, but the Bevy _editor_ has the goal of being useful to production game studios which IIUC absolutely need the ability for their non-programmer team members to work without touching or knowing code. Rest assured the Bevy devs are keeping your use case in mind as they discuss the plans for the editor!

So the work with the editor has not started yet? When will it start?

**Reply by colepoirier** [reply]
I believe work on the editor is planned to start by the end of 2022. Currently, there are a number of 3rd party editor prototypes including bevy_editor_pls, and bevy_blender_toolkit.

**Reply by C-BJ** [reply]
> I believe work on the editor is planned to start by the end of 2022. Currently, there are a number of 3rd party editor prototypes including bevy_editor_pls, and bevy_blender_toolkit.

Thanks
